Relocate store.server.js and related files/tests.

mojit0 opened the pull request:
Moved store.server.js and package-walker.server.js to app/autoload where they're more appropriate.
Added new store.js file at the top level, and updated mojito.js to rely on it to configure the store instance.
Migrated all tests related to the relocated files to their proper location in the test hierarchy.
Patched a minor glitch (no mojito-assets-addon in requires[]) in the default archetypes for mojit which caused new mojits to fail on startup.
Patched start.js command output to avoid using localhost and instead to rely on 127.0.0.1 which works with or without an /etc/hosts entry or similar mapping for localhost.

mojit0 committed:
Fix localhost reference.

With new startup logic the proper reference is 127.0.0.1 so you can click on the log entry and have it work consistently.

mojit0 committed:
Fix buried error message.

The true error is masked here. Repaired to always report actual error, which can come from an inadequately configured sandbox object etc.

mojit0 committed:
Fix missing assets addon in mojit archetypes.

The assets add-on is needed for the addCss call in the default and full archetypes, but isn't included by default so simple new mojits will fail.

mojit0 committed:
Refactor initial resource store creation.

Migrated the initial logic for resource store creation out of the mojito.js file and into a store.js file. This secondary file creates the  YUI sandbox and store, getting it ready for use.

mojit0 committed:
Remove debugging console call.

mojit0 committed:
Relocate store.server.js to app/autoload.

Moved store.server.js and related package-walker.server.js to the app/autoload directory.
Adjusted store.server.js to have better defaulting when config isn't complete or isn't provided.
Repaired sandbox limitations in the yui.server.js add-on (require and module aren't part of default vm context).

mojit0 committed:
Add missing file.

mojit0 committed:
Remove debug logging calls.

mojit0 committed:
Remove lint.

mojit0 committed:
Migrate test files for relocated lib files.

Moved the store.server.js and package-walker.server.js files so we're relocating their respective test files and descriptors.

mojit0 committed:
Fix broken test.

The configuration test data was out of sync with the fixture application.json content so the test was failing.

caridy commented:
We will also need to update all the commands since most of them uses the store. This is kind of challenging because they might also use YUI (in a very minimum way) but we might need to do some clean up I think.

mojit0 commented:
@caridy re: the commands, yes. I'll do that today, along with untangling the conflict re: Drew's unit test fixes, and update the PR.

caridy commented:
+1

we can refine the factory later on.

mojit0 committed:
Merge upstream changes.

mojit0 committed:
Merge remote-tracking branch 'upstream/develop-perf' into develop-perf

mojit0 committed:
Fix buried error messages when loading commands.

The true error message for a poorly built command is never reported, only that the command is "Invalid" which is misleading at the very least.

mojit0 committed:
Fix remaining resource store and add-on references.
